app/src/components/Map/SearchPanel.jsx

```diff
@@ -1,84 +1,108 @@
     };
+  }
 
-    this.toggleVisibleAdvancedSearch = this.toggleVisibleAdvancedSearch.bind(this);
-    this.onSearchInputChange = this.onSearchInputChange.bind(this);
+  componentWillReceiveProps(nextProps) {
+    if (!!nextProps.search) {
+      Object.assign(this.state.search, nextProps.search);
+      this.setState(this.state);
+    }
   }
 
   onSearchInputChange(event) {
-    this.props.getSearchResults(event.target.value);
-  }
+    this.keyword = event.target.value;
 
-  toggleVisibleAdvancedSearch() {
-    this.setState({
-      visibleAdvancedSearch: !this.state.visibleAdvancedSearch
+    Object.assign(this.state, {
+      isEmpty: !this.keyword.length > 0,
+      firsTime: false
     });
+
+    this.setState(this.state);
+
+    if (this.keyword.length < 3) return;
+
+    this.props.getSearchResults(this.keyword);
+  }
+
+  cleanResults() {
+    this.state.search.entries = [];
+    this.setState(this.state);
   }
 
   render() {
-    let advancedSearchAccordion = (
-      <Accordion
-        allowMultiple={false}
-        activeItems={6}
-        onChange={this.toggleVisibleAdvancedSearch}
-      >
-        <AccordionItem
-          title={this.state.visibleAdvancedSearch ? 'hide advanced search' : 'advanced search'}
-          key="advancedsearch"
-          titleClassName={controlPanelStyle.text_search}
-        >
-          <div>
-            <FiltersPanel />
-            <span className={controlPanelStyle.button_advanced_search}>SEARCH</span>
-          </div>
-        </AccordionItem>
-      </Accordion>
-    );
-
-    let searchResults = [];
-    if (this.props.search && !this.state.visibleAdvancedSearch) {
-      for (let i = 0, length = this.props.search.entries.length; i < length; i++) {
+    const searchResults = [];
+
+    if (this.state.search) {
+      for (let i = 0, length = this.state.search.entries.length; i < length; i++) {
         searchResults.push(
-          <li
+          <SearchResult
+            className={searchPanelStyles.result}
             key={i}
-            onClick={() => this.props.drawVessel(this.props.search.entries[i])}
-          >
-            {this.props.search.entries[i].vesselname},<span>{this.props.search.entries[i].mmsi}</span>
-          </li>
+            keyword={this.keyword}
+            drawVessel={this.props.drawVessel}
+            vesselInfo={this.state.search.entries[i]}
+            setVesselPosition={this.props.setVesselPosition}
+            toggleVisibility={this.props.toggleVisibility}
+            vesselVisibility={this.props.vesselVisibility}
+          />
         );
       }
     }
 
     return (
-      <div>
+      <div className={searchPanelStyles['c-search-panel']}>
         <input
-          onChange={this.onSearchInputChange}
-          className={controlPanelStyle.input_accordion}
-          placeholder="Type your search query"
+          id="search-vessels"
+          type="text"
+          onChange={(e) => this.onSearchInputChange(e)}
```

Comment:
I think we'll generate way too much queries with that. Can't we have a search button or at least debounce the call?